pkg/apis/api/v1alpha1/astarte_types.go

```diff
@@ -242,6 +242,33 @@ type AstarteRabbitMQSpec struct {
 	EventsExchangeName string `json:"eventsExchangeName,omitempty"`
 }
 
+type AstarteCassandraSSLConfigurationSpec struct {
+	Enabled bool `json:"enabled"`
+	// +optional
+	CustomCASecret v1.LocalObjectReference `json:"customCASecret,omitempty"`
+	// +optional
+	SNI *bool `json:"sni,omitempty"`
+	// +optional
+	CustomSNI string `json:"customSNI,omitempty"`
+}
+
+type LoginCredentialsSecret struct {
+	Name        string `json:"name"`
+	UsernameKey string `json:"usernameKey"`
+	PasswordKey string `json:"passwordKey"`
+}
+
+type AstarteCassandraConnectionSpec struct {
+	// +optional
+	PoolSize *int `json:"poolSize,omitempty"`
+	// +optional
+	Autodiscovery *bool `json:"autodiscovery,omitempty"`
+	// +optional
+	SSLConfiguration AstarteCassandraSSLConfigurationSpec `json:"sslConfiguration,omitempty"`
+	// +optional
+	Secret *LoginCredentialsSecret `json:"secret,omitempty"`
```

Comment:
Just like in RabbitMQ, you should still provide a way to give out the Username+Password directly, without the secret.